app/javascript/components/Product/index.tsx

```diff
@@ -253,9 +253,17 @@ export const Product = ({
 
   const notForSaleMessage = getNotForSaleMessage(product);
   const [discountCode, setDiscountCode] = React.useState(initialDiscountCode);
+
+  React.useEffect(() => {
+    setDiscountCode(initialDiscountCode);
+  }, [initialDiscountCode]);
+
   const selectionAttributes = applySelection(product, discountCode?.valid ? discountCode.discount : null, selection);
   let { basePriceCents } = selectionAttributes;
   const { priceCents, discountedPriceCents, pppDiscounted, isPWYW, maxQuantity } = selectionAttributes;
+  // Multiply by quantity to show total price when quantity > 1
+  const totalBasePriceCents = basePriceCents * selection.quantity;
+  const totalDiscountedPriceCents = discountedPriceCents * selection.quantity;
```

Comment:
Do we need this? we have these functionalities already when we apply a discount through URL, right?